salt/modules/cassandra_cql.py

```diff
@@ -0,0 +1,585 @@
+
+
+def _query(query, contact_points=None, port=None, cql_user=None, cql_pass=None):
+    '''
+    Run a query on a Cassandra cluster and return a dictionary.
+
+    :param query:          The query to execute.
+    :type  query:          str
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :param params:         The parameters for the query, optional.
+    :type  params:         str
+    :return:               A dictionary from the return values of the query
+    :rtype:                list[dict]
+    '''
+    cluster, session = _connect(contact_points=contact_points, port=port, cql_user=cql_user, cql_pass=cql_pass)
+
+    session.row_factory = dict_factory
+    ret = []
+
+    try:
+        results = session.execute(query)
+    except BaseException as e:
+        log.error('Something went wrong when executing the following query: %s', query)
+        msg = "ERROR: Cassandra query failed: %s" % query
+        raise CommandExecutionError(msg, e)
+
+    if results:
+        for index in range(0, len(results)):
+            result = results[index]
+            values = {}
+            for key, value in result.iteritems():
+                # Salt won't return dictionaries with odd types like uuid.UUID
+                if not isinstance(value, unicode):
+                    value = str(value)
+                values[key] = value
+            ret.append(values)
+    cluster.shutdown()
+    return ret
+
+
+def version(contact_points=None, port=None, cql_user=None, cql_pass=None):
+    '''
+    Show the Cassandra version.
+
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :return:               The version for this Cassandra cluster.
+    :rtype:                str
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt 'minion1' cassandra.version
+
+        salt 'minion1' cassandra.version contact_points=minion1
+    '''
+    query = 'select release_version from system.local limit 1'
+    ret = _query(query, contact_points, port, cql_user, cql_pass)
+    if ret:
+        return ret[0].get('release_version')
+    else:
+        return False
```

Comment:
This is inconsistent return. Here you return `False`, but in function `info` below you chose to return `None`. In case you want to return always `None`, this block I would write like this:
```python
query = ...
ret = _query(...
return ret and ret[0].get('release_version') # or False
```